discovery/api/server/api.go

```diff
@@ -77,9 +78,19 @@ func (w *Wrapper) GetPresentations(ctx context.Context, request GetPresentations
 }
 
 func (w *Wrapper) RegisterPresentation(ctx context.Context, request RegisterPresentationRequestObject) (RegisterPresentationResponseObject, error) {
-	err := w.Server.Register(ctx, request.ServiceID, *request.Body)
+	err := w.Server.Register(contextWithForwardedHost(ctx), request.ServiceID, *request.Body)
 	if err != nil {
 		return nil, err
 	}
 	return RegisterPresentation201Response{}, nil
 }
+
+func contextWithForwardedHost(ctx context.Context) context.Context {
+	// cast context to echo.Context
+	echoCtx := ctx.Value("echo.Context")
+	if echoCtx != nil {
+		// forward X-Forwarded-Host header via context
+		ctx = context.WithValue(ctx, "X-Forwarded-Host", echoCtx.(echo.Context).Request().Header.Get("X-Forwarded-Host"))
```

Comment:
you shouldn't use string values for context keys, but unexported structs (codeclimate will complain)